evap/evaluation/tests/tools.py

```diff
@@ -341,6 +346,91 @@ def setUpClass(cls) -> None:
         cls.selenium.set_window_size(*cls.window_size)
 
 
+@tag("vrt")
+class VisualRegressionTestCase(LiveServerTest):
+    window_size = (1920, 1080)
+    _http_timeout_seconds = 3
+
+    def setUp(self) -> None:
+        with connection.cursor() as cursor:
+            # reset userprofile id sequence to keep test reproducible
+            cursor.execute("ALTER SEQUENCE evaluation_userprofile_id_seq restart")
+
+        super().setUp()
+
+    @property
+    def viewport(self):
+        return f"{self.window_size[0]}x{self.window_size[1]}"
+
+    def trigger_screenshot(self, name: str):
+        full_name = self.__class__.__name__ + "_" + name
+
+        config = {
+            "apiUrl": os.environ.get("VRT_APIURL"),
+            "ciBuildId": os.environ.get("VRT_CIBUILDID"),
+            "branchName": os.environ.get("VRT_BRANCHNAME"),
+            "apiKey": os.environ.get("VRT_APIKEY"),
+            "projectId": os.environ.get("VRT_PROJECT"),
+        }
+
+        headers = {
+            "apiKey": config.get("apiKey"),
+            "Content-Type": "application/json",
+        }
+
+        data = {
+            "project": config.get("projectId"),
+            "branchName": config.get("branchName"),
+            "ciBuildId": config.get("ciBuildId"),
+        }
+
+        registration_response = requests.post(
+            f'{config.get("apiUrl")}/builds',
+            data=json.dumps(data),
+            headers=headers,
+            timeout=self._http_timeout_seconds,
+        )
+
+        registration_response.raise_for_status()
+
+        test_data = {
+            "project": config.get("projectId"),
+            "projectId": config.get("projectId"),
+            "branchName": config.get("branchName"),
+            "ciBuildId": config.get("ciBuildId"),
+            "name": full_name,
+            "imageBase64": self.selenium.get_screenshot_as_base64(),
+            "viewport": self.viewport,
+            "buildId": registration_response.json().get("id"),
+        }
+
+        test_response = requests.post(
+            f'{config.get("apiUrl")}/test-runs',
+            data=json.dumps(test_data),
+            headers=headers,
+            timeout=self._http_timeout_seconds,
+        )
+
+        test_response.raise_for_status()
+
+        review_url = test_response.json().get("url", "<url-not-found>")
+        switcher = {
+            "new": f"No Baseline! Review manually: {review_url}",
+            "unresolved": f"Difference found: {review_url}",
+        }
+
+        error_message = switcher.get(test_response.json().get("status"))
+
+        _ = requests.patch(
```

Comment:
what are we doing here? does it reset the state to what was the baseline before the test? if so, this should be more robust with try-finally or similar